devdoc/interlocked_win32_requirements.md

```diff
@@ -0,0 +1,346 @@
+MOCKABLE_FUNCTION(, int64_t, interlocked_xor_64, volatile int64_t*, destination, int64_t, value);
+MOCKABLE_FUNCTION(, int8_t, interlocked_xor_8, volatile int8_t*, destination, int8_t, value);
+```
+
+## interlocked_add
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_add, volatile int32_t*, addend, int32_t, value);
+```
+**SRS_INTERLOCKED_WIN32_43_001 [** `interlocked_add` shall call `InterlockedAdd` from `windows.h`. **]**
+
+**SRS_INTERLOCKED_WIN32_43_002: [** `interlocked_add` shall return the resulting 32-bit integer. **]**
+
+## interlocked_and
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_and, volatile int32_t*, destination, int32_t, value);
+```
+**SRS_INTERLOCKED_WIN32_43_003 [** `interlocked_and` shall call `InterlockedAnd` from `windows.h`. **]**
+
+**SRS_INTERLOCKED_WIN32_43_004: [**`interlocked_and` shall return the resulting 32-bit integer. **]**
+
+## interlocked_and_16
+
+```c
+MOCKABLE_FUNCTION(, int16_t, interlocked_and_16, volatile int16_t*, destination, int16_t, value);
+```
+**SRS_INTERLOCKED_WIN32_43_005 [** `interlocked_and_16` shall call `InterlockedAnd16` from `windows.h`. **]**
+
+**SRS_INTERLOCKED_WIN32_43_006: [** `interlocked_and_16` shall return the resulting 16-bit integer. **]**
+ 
+ ## interlocked_and_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_and_64, volatile int64_t*, destination, int64_t, value);
+```
+**SRS_INTERLOCKED_WIN32_43_007 [** `interlocked_and_64` shall call `InterlockedAnd64` from `windows.h`. **]**
+
+**SRS_INTERLOCKED_WIN32_43_008: [** `interlocked_and_64` shall return the resulting 64-bit integer. **]**
+
+## interlocked_and_8
+
+```c
+MOCKABLE_FUNCTION(, int8_t, interlocked_and_8, volatile int8_t*, destination, int8_t, value);
+```
+**SRS_INTERLOCKED_WIN32_43_009 [** `interlocked_and_8` shall call `InterlockedAnd8` from `windows.h`. **]**
+
+**SRS_INTERLOCKED_WIN32_43_010: [** `interlocked_and_8` shall return the resulting 8-bit integer. **]**
+
+## interlocked_compare_exchange
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_compare_exchange, volatile int32_t*, destination, int32_t, exchange, int32_t, comperand);
+```
+**SRS_INTERLOCKED_WIN32_43_011 [** `interlocked_compare_exchange` shall call `InterlockedCompareExchange` from `windows.h`. **]**
+
+**SRS_INTERLOCKED_WIN32_43_012: [** `interlocked_compare_exchange` shall return the initial value of the `destination` parameter. **]**
+
+## interlocked_compare_exchange_128
+
+```c
+MOCKABLE_FUNCTION(, bool, interlocked_compare_exchange_128, volatile int64_t*, destination, int64_t, exchange_high, int64_t, exchange_low, int64_t*, comperand_result);
+```
+**SRS_INTERLOCKED_WIN32_43_013 [** `interlocked_compare_exchange_128` shall call `interlocked_compare_exchange_128` from `windows.h`. **]**
+
+**SRS_INTERLOCKED_WIN32_43_014: [** `interlocked_compare_exchange_128` shall return 1 if `comperand_result` equals the original value of the `destination` parameter, 0 if otherwise.**]**
```

Comment:
>`interlocked_compare_exchange_128` shall return 1 if `comperand_result` equals the original value of the `destination` parameter, 0 if otherwise [](start = 36, length = 144)

I would also split this and maybe re-word:

If `InterlockedCompareExchange128` returns 0 then `interlocked_compare_exchange_128` shall return `false`.

Otherwise  `interlocked_compare_exchange_128` shall return `true`.